packages/stream-json/reader.js

```diff
@@ -0,0 +1,26 @@
+// @ts-check
+/// <reference types="ses"/>
+
+/* Adapts a Reader<Uint8Array> to a JSON object Reader<any> where reader and
+ * writer streeams are modeled as hybrid async iterators + generators.
+ */
+const textDecoder = new TextDecoder();
+
+/**
+ * @param {import('@endo/stream').Reader<Uint8Array, undefined>} reader
+ */
+async function* makeJsonAsyncIterator(reader) {
+  for await (const bytes of reader) {
+    const text = textDecoder.decode(bytes);
+    yield JSON.parse(text);
+  }
+}
+
+/**
+ * @param {import('@endo/stream').Reader<Uint8Array, undefined>} reader
+ * @returns {import('@endo/stream').Reader<ReturnType<JSON.parse>, undefined>}
```

Comment:
Afaik `JSON.parse` is types to return `any`, right?

It might be more useful(but possibly unsafe) to provide a generic version here.

Reply:
I was not so sure because the signature is theoretically narrower than that. I figured stating this would give the types room to evolve.